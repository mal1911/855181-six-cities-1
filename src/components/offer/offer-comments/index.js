import OfferComments from './offer-comments';

export default OfferComments;
