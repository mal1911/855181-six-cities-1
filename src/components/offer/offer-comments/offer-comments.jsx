import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {offerType, commentType} from "../../../prop-types";
import {connect} from "react-redux";
import {Operation} from "../../../reducer/data/data";
import {getCommentsData} from "../../../reducer/data/selectors";
import {getAuthorizationStatus} from "../../../reducer/user/selectors";
import OfferComment from "../offer-comment";
import OfferForm from "../offer-form";
import {MAX_COMMENTS} from "../../../constants";

class OfferComments extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.onLoaded(this.props.offerObj.id);
  }

  render() {
    const {commentsData, isAuthorizationRequired, offerObj} = this.props;
    const comments = commentsData.slice(0, MAX_COMMENTS).map((commentObj, index) =>
      <OfferComment key={index} commentObj={commentObj}/>);

    const withOfferForm = !isAuthorizationRequired ? <OfferForm offerId={offerObj.id}/> : null;

    return <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{commentsData.length}</span></h2>
      <ul className="reviews__list">
        {comments}
      </ul>
      {withOfferForm}
    </section>;
  }
}

OfferComments.propTypes = {
  commentsData: PropTypes.arrayOf(commentType),
  offerObj: offerType,
  onLoaded: PropTypes.func,
  isAuthorizationRequired: PropTypes.bool,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  commentsData: getCommentsData(state),
  isAuthorizationRequired: getAuthorizationStatus(state),
});

const mapDispatchToProps = (dispatch) => ({
  onLoaded: (id) => {
    dispatch(Operation.loadCommentsData(id));
  },
});

export {OfferComments};

export default connect(mapStateToProps, mapDispatchToProps)(OfferComments);
