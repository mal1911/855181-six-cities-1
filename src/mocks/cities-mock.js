export const citiesMock = [
  {
    id: 1,
    name: `Paris`,
  },
  {
    id: 2,
    name: `Cologne`,
  },
  {
    id: 3,
    name: `Brussels`,
  },
  {
    id: 4,
    name: `Amsterdam`,
  },
  {
    id: 5,
    name: `Hamburg`,
  },
  {
    id: 6,
    name: `Dusseldorf`,
  },
];

