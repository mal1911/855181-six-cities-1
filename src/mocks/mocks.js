export const offersData = [
  {
    id: 1,
    isPremium: true,
    city: `Paris`,
    title: `Paris Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`,
    coordinates: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 2,
    isPremium: false,
    city: `Paris`,
    title: `Paris Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`,
    coordinates: [52.369553943508, 4.85309666406198],
  },
  {
    id: 3,
    isPremium: false,
    city: `Paris`,
    title: `Paris Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 4,
    isPremium: false,
    city: `Paris`,
    title: `Paris Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 5,
    isPremium: true,
    city: `Cologne`,
    title: `Cologne Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 6,
    isPremium: false,
    city: `Cologne`,
    title: `Cologne Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 7,
    isPremium: true,
    city: `Brussels`,
    title: `Brussels Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 8,
    isPremium: false,
    city: `Brussels`,
    title: `Brussels Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 9,
    isPremium: true,
    city: `Amsterdam`,
    title: `Amsterdam Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 10,
    isPremium: false,
    city: `Amsterdam`,
    title: `Amsterdam Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 11,
    isPremium: true,
    city: `Hamburg`,
    title: `Hamburg Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 12,
    isPremium: false,
    city: `Hamburg`,
    title: `Hamburg Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
  {
    id: 13,
    isPremium: true,
    city: `Dusseldorf`,
    title: `Dusseldorf Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 14,
    isPremium: false,
    city: `Dusseldorf`,
    title: `Dusseldorf Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`,
    coordinates: [52.3809553943508, 4.939309666406198],
  },
];

export const citiesData = [
  {
    name: `Paris`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Cologne`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Brussels`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Hamburg`,
    coordinates: [52.38333, 4.9],
  },
  {
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9],
  },
];
