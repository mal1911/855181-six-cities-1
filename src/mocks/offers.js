export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    price: 10,
    rating: 1,
    type: `Apartment`,
    imgSrc: `img/apartment-01.jpg`
  },
  {
    title: `Wood and stone place`,
    price: 20,
    rating: 2,
    type: `Apartment`,
    imgSrc: `img/apartment-02.jpg`
  },
  {
    title: `Canal View Prinsengracht`,
    price: 50,
    rating: 4,
    type: `Apartment`,
    imgSrc: `img/apartment-03.jpg`
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    price: 100,
    rating: 5,
    type: `Private room`,
    imgSrc: `img/room.jpg`
  },
];
