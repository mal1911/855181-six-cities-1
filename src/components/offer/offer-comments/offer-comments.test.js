import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {OfferComments} from "./offer-comments";
import {offersData, commentsData, initialState} from "../../../mocks/mocks";
import {BrowserRouter} from "react-router-dom";

describe(`OfferComments correctly renders`, () => {
  const mockStore = configureStore();
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore(initialState);
    tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <OfferComments
              commentsData={commentsData}
              offerObj={offersData[0]}
              onLoaded={jest.fn()}
              isAuthorizationRequired={false}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();
  });

  it(`OfferComments correctly renders`, () => {
    expect(tree).toMatchSnapshot();
  });
});
