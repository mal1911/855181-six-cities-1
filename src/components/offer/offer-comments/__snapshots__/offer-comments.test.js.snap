// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferComments correctly renders OfferComments correctly renders 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      3
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://es31-server.appspot.com/six-cities/static/avatar/6.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Kendall
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-24"
        >
          June 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://es31-server.appspot.com/six-cities/static/avatar/5.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Corey
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Home is amazing. It's like staying in a museum. The rooms, furnishings and artworks are incredible. The views of My Vesuvius
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-24"
        >
          June 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://es31-server.appspot.com/six-cities/static/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Mollie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Beautiful space, fantastic location and atmosphere, really a wonderful place to spend a few days. Will be back.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-24"
        >
          June 2019
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        id="1-star"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-star"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      id="review"
      maxLength={300}
      minLength={50}
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value=""
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set 
        <span
          className="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
